cuda_bindings/cuda/bindings/cydriver.pyx.in

```diff
@@ -1595,8 +1658,35 @@ cdef CUresult cuLaunchCooperativeKernelMultiDevice(CUDA_LAUNCH_PARAMS* launchPar
 
 {{if 'cuLaunchHostFunc' in found_functions}}
 
+ctypedef struct cuHostCallbackData_st:
+    CUhostFn callback
+    void *userData
+
+ctypedef cuHostCallbackData_st cuHostCallbackData
+
+@cython.show_performance_hints(False)
+cdef void cuHostCallbackWrapper(void *data) nogil:
```

Comment:
Once we move this wrapper to the Python layer, we can define it as `with gil` in the function signature, then we don't need the `with gil` block in the function body. The GIL can be acquired at function call time, and Cython has a bit more info for the GIL analysis.
https://cython.readthedocs.io/en/latest/src/userguide/nogil.html#releasing-and-reacquiring-the-gil